MSAL/src/native_auth/network/errors/reset_password/MSALNativeAuthResetPasswordPollCompletionResponseError.swift

```diff
@@ -83,6 +83,7 @@ extension MSALNativeAuthResetPasswordPollCompletionResponseError {
              .expiredToken,
              .invalidRequest,
              .userNotFound,
+             .unknownCase,
              .none:
```

Comment:
Does it still make sense to manage the ".none" case for the error field? At this point can makes more sense to just don't the error field as optional

Reply:
As discussed, I've:

- Made the `error` field non-optional
- Added a test where the `error` field is missing (see `MSALNativeAuthResponseErrorHandlerTests`)
- Modified the expected behavior of two tests (see `MSALNativeAuthResponseErrorHandlerTests`)